src/app/dataset-create/dataset-create.component.ts

```diff
@@ -1,9 +1,135 @@
-import { ChangeDetectionStrategy, Component } from "@angular/core";
+import { BaseComponent } from "src/app/common/base.component";
+/* eslint-disable @typescript-eslint/unbound-method */
+import { DatasetKind } from "src/app/api/kamu.graphql.interface";
+import { MaybeNull } from "./../common/app.types";
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    OnInit,
+} from "@angular/core";
+import * as monaco from "monaco-editor";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { AppDatasetCreateService } from "./dataset-create.service";
 
 @Component({
     selector: "app-dataset-create",
     templateUrl: "./dataset-create.component.html",
     styleUrls: ["./dataset-create.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DatasetCreateComponent {}
+export class DatasetCreateComponent extends BaseComponent implements OnInit {
+    public yamlTemplate: MaybeNull<string> = "";
+    public fileName: MaybeNull<string> = "";
+    public showMonacoEditor = false;
+    public errorMessage = "";
+    public initialHint = "# You can edit this file\n";
+    public owners = ["kamu"];
+    public sqlEditorOptions: monaco.editor.IStandaloneEditorConstructionOptions =
+        {
+            theme: "vs",
+            language: "yaml",
+            renderLineHighlight: "none",
+            minimap: {
+                enabled: false,
+            },
+        };
+    public createDatasetForm: FormGroup = this.fb.group({
+        owner: ["kamu", [Validators.required]],
+        datasetName: ["", [Validators.required]],
+        kind: ["root", [Validators.required]],
+    });
+    private kindMapper: Record<string, DatasetKind> = {
+        root: DatasetKind.Root,
+        derivative: DatasetKind.Derivative,
+    };
+
+    constructor(
+        private cdr: ChangeDetectorRef,
+        private fb: FormBuilder,
+        private datasetCreateService: AppDatasetCreateService,
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.trackSubscription(
+            this.datasetCreateService.onErrorMessageChanges.subscribe(
+                (message: string) => {
+                    this.errorMessage = message;
+                    this.cdr.detectChanges();
+                },
+            ),
+        );
+    }
+
+    public get isFormValid(): boolean {
+        if (this.yamlTemplate) return true;
+        return this.createDatasetForm.valid;
+    }
+
+    public onCreateDataset(): void {
+        this.showMonacoEditor
+            ? this.createDatasetFromSnapshot()
+            : this.createDatasetFromForm();
+    }
+
+    public onFileSelected(event: Event): void {
+        this.yamlTemplate = "";
```

Comment:
Maybe clean existing after we checked there is some file?

Reply:
Yes, better after checked